# Test DecisionTreeClassifier:

#### 1. Use the iris.csv dataset

In [1]:
from si.io.csv_file import read_csv
import pandas as pd

In [2]:
df_path = r"C:\Users\Mariana\si\datasets\iris\iris.csv"
df = read_csv(df_path, sep = ',', features = True, label = True) 

#### 2. Split the data train and test

In [3]:
from si.model_selection.split import train_test_split
from si.models.decision_tree_classifier import DecisionTreeClassifier

In [4]:
train_dataset, test_dataset = train_test_split(df, test_size=0.33, random_state=42)

In [9]:
print(f'(As dimensões dos dados de treino correspondem a:', train_dataset.shape())
print(f'(As dimensões dos dados de teste correspondem a:',test_dataset.shape())

(As dimensões dos dados de treino correspondem a: (101, 4)
(As dimensões dos dados de teste correspondem a: (49, 4)


Os dados de treino apresentam 101 linhas e 4 colunas, enquanto que os dados de teste apresentam 49 linhas e 4 colunas.

#### 3. Train a Decision Tree Classifier using the training data

In [10]:
model = DecisionTreeClassifier(min_sample_split=2, max_depth=5, mode='gini')
model.fit(train_dataset)

#### 4. Print the tree

In [11]:
model.print_tree()

petal_length <= 1.9
	left: Iris-setosa
	right: petal_width <= 1.7
	  left: petal_length <= 5.1
	    left: sepal_length <= 4.9
	      left: sepal_width <= 2.4
	        left: Iris-versicolor
	        right: Iris-virginica
	      right: sepal_width <= 2.2
	        left: petal_length <= 4.0
	          left: Iris-versicolor
	          right: Iris-virginica
	        right: Iris-versicolor
	    right: Iris-virginica
	  right: petal_length <= 4.8
	    left: sepal_length <= 5.9
	      left: Iris-versicolor
	      right: Iris-virginica
	    right: Iris-virginica


#### 5. Evaluate the model on the testing data

In [12]:
print(model.score(test_dataset))

0.9591836734693877


O score de 0.95918 indica que o modelo obteve uma precisão de cerca de 95.9% no conjunto de dados de teste. Este valor sugere que o modelo utilizado é bastante bom para a classificação no conjunto de teste.